python/lsst/sims/utils/stellarMags.py

```diff
@@ -71,34 +74,47 @@ def stellarMags(stellarType, rmag=19.):
-                ['F','kp00_7250[g45]',0.9140316091,0.1254277486,-0.03419150003,-0.0802010739,-0.03802756413],
-                ['G','kp00_6000[g45]',1.198219095,0.3915608688,0.09129426676,0.002604263747,-0.004659443668],
-                ['K','kp00_5250[g45]',1.716635024,0.6081567546,0.1796910856,0.06492278686,0.0425155827],
-                ['M','kp00_3750[g45]',2.747842719,1.287599638,0.5375622482,0.4313486709,0.219308065],
-                ['HeWD_25200_80', 'bergeron_He_24000_80.dat_25200.gz', -0.218959, -0.388374, -0.326946, -0.253573, -0.239460],
-                ['WD_11000_85', 'bergeron_10500_85.dat_11000.gz', 0.286146, -0.109115, -0.178500, -0.185833, -0.186913],
-                ['WD_3000_85', 'bergeron_2750_85.dat_3000.gz', 3.170620, 1.400062, 0.167195, 0.127024, -0.378069]],
-                dtype=zip(names,types))
+            ['O', 'kp00_50000[g50]', -0.4835688497, -0.5201721327, 
+             -0.3991733698, -0.3106800468, -0.2072290744],
+            ['B', 'kp00_30000[g40]', -0.3457202828, -0.4834762052, 
+             -0.3812792176, -0.2906072887, -0.1927230035],
+            ['A', 'kp00_9500[g40]', 0.8823182684, -0.237288029, 
+             -0.2280783991, -0.1587960264, -0.03043824335],
+            ['F', 'kp00_7250[g45]', 0.9140316091, 0.1254277486, 
+             -0.03419150003, -0.0802010739, -0.03802756413],
+            ['G', 'kp00_6000[g45]', 1.198219095, 0.3915608688,
+             0.09129426676, 0.002604263747, -0.004659443668],
+            ['K', 'kp00_5250[g45]', 1.716635024, 0.6081567546,
+             0.1796910856, 0.06492278686, 0.0425155827],
+            ['M', 'kp00_3750[g45]', 2.747842719, 1.287599638,
+             0.5375622482, 0.4313486709, 0.219308065],
+            ['HeWD_25200_80', 'bergeron_He_24000_80.dat_25200.gz', 
+             -0.218959, -0.388374, -0.326946, -0.253573, -0.239460],
+            ['WD_11000_85', 'bergeron_10500_85.dat_11000.gz',
+             0.286146, -0.109115, -0.178500, -0.185833, -0.186913],
+            ['WD_3000_85', 'bergeron_2750_85.dat_3000.gz',
+             3.170620, 1.400062, 0.167195, 0.127024, -0.378069]],
+            dtype=list(zip(names, types)))
         # Switch to a dict for faster look-up
-        stellarMags.data={}
+        stellarMags.data = {}
         for row in data:
-            stellarMags.data[row['stellarType']] = row
+            stellarMags.data['%s' % row['stellarType']] = row
 
     results = {}
-    #good = np.where(stellarMags.data['stellarType'] == stellarType)
-    if stellarType not in stellarMags.data.keys():
-        raise ValueError('stellarType must be one of '+', '.join(stellarMags.data.keys()))
+    # good = np.where(stellarMags.data['stellarType'] == stellarType)
+    if stellarType not in stellarMags.data:
+        message = 'Received stellarType %s' % stellarType
+        message += ' but expected one of %s' % ', '.join(list(stellarMags.data.keys()))
```

Comment:
Yes, you're right. I wasn't sure if the generator would work quite the same way, but it looks like it does.